### ISSUE TITLE
onPageLoad od onDomReady

### ISSUE THREAD

**Issue by danieleleoni** [jasonslyvia/react-lazyload#127]
Hi guys,

I see issues regarding componentDidMount and so on, but I did not found anything anything about the possibility to defer element load not only on presence into viewport, but also after DOM is ready or page is fully loaded.

Is there anything I miss?
Into https://jasonslyvia.github.io/react-lazyload/examples/#/fadein it seems to defer images, but I can't understand how.

It should be very useful for us, since our page has an image on the top part of the screen

Thanks!!

**Comment by jasonslyvia** [comment]
I believe this is somehow beyond the responsibility of `react-lazyload`, it is up to you to make sure your app is rendered after DOM ready or page load.

**Comment by danieleleoni** [comment]
I do not agree completely, because one of the scope of `react-lazyload` is (tell me if I'm wrong) to prevent application to load images all at once and reduce page load time. Isn't it?
Do you use this package in combination with others to solve the issue eventually ?

**Comment by jasonslyvia** [comment]
By saying `lazy load`, in frontend development context we mean to load stuff only if they appear in viewport in order to improve page performance and reduce unwanted networking expense.

What's your specific usage that might render a React App before DOM ready?

**Comment by danieleleoni** [comment]
We have a SSR application and we focus on page speed.
We just want to serve important content as soon as possible and images after that.

<img width="1392" alt="screen shot 2017-10-13 at 09 59 03" src="https://user-images.githubusercontent.com/20039309/31535969-4d1e8dd4-affd-11e7-99ee-96739410c892.png">

To be clear we would like to load images after blue line (or red even better)

**Comment by jasonslyvia** [comment]
If you're using SSR, there is even no such DOM event (`DOMContentLoaded` or `load`) and react-lazyload will just render its children, so that will be your issue in this case?

**Comment by danieleleoni** [comment]
Even with a Ssr rendering there are elements that should be rendered only
on client and only when dom is ready.

My problem is exactly what you see in the screenshot, I would like to load
assets later when page is ready

**Comment by lidawang** [comment]
Sounds like what you're asking for is a new feature - I use this library in a SSR webapp as well, but we prefer our images (if they're in the viewport) to be loaded ASAP so the user doesn't see it come in later and make pagespeed (aka perceived page load time) worse. This does sound implementable, but I'm not sure it's the right thing to do.

**Comment by jasonslyvia** [comment]
I agree, sure we can add some kind of `document.readyState === 'complete'` check, but is that necessary? will this behavior make others confused?

**Comment by danieleleoni** [comment]
It should be an option so it does not change the default behaviour anyway.

**Comment by GiancarlosIO** [comment]
I'm agree with @danieleleoni, I have the same use-case, load images after the dom is full-loaded to improve the pagespeed.
